gpui: update dependencies

niklaswimmer opened the pull request:
This PR updates dependencies with the goal to unify transitive dependency's versions. The impact does not seem to be as high as I originally hoped, but a Hyperfine run still showed a 13% improvement over main when compiling just gpui (testing with the whole Zed build would have been to time-consuming).

<details>
<summary>Hyperfine benchmark</summary>
<br/>

```bash
CARGO_TARGET_DIR=~/.cargo/.target/zed-timing hyperfine -w 2 -r 8 -L branch main,gpui/linux/deps-cleanup -s 'git switch {branch}' -p 'cargo clean' 'cargo build -p gpui'
```

![image](https://github.com/zed-industries/zed/assets/51296580/9185e375-479c-489f-9dde-9144c3d0a542)
</details>

<details>
<summary>Cargo tree diff</summary>
<br/>

```bash
cargo tree -d  --format '{p}: {f}' -e normal,build --depth 0 | sed -r '/^\s*$/d'
```

This command lists all duplicate dependencies (and then strips new lines). The result of diffing the output of main against this branch is attached below.

```diff
1,4d0
< arrayvec v0.5.2: 
< arrayvec v0.7.4: default,std
< async-broadcast v0.5.1: 
< async-broadcast v0.7.0: 
25,28d20
< clap v3.2.25: atty,color,default,std,strsim,suggestions,termcolor
< clap v4.4.4: color,default,derive,error-context,help,std,suggestions,usage
< clap_lex v0.2.4: 
< clap_lex v0.5.1: 
42,43d33
< fontdb v0.5.4: fs,memmap2
< fontdb v0.15.0: fontconfig,fontconfig-parser,fs,memmap,memmap2,std
45a36,37
> gif v0.12.0: color_quant,default,raii_no_panic,std
> gif v0.13.1: color_quant,default,raii_no_panic,std
53d44
< indexmap v1.9.3: std
59a51,52
> kurbo v0.9.5: default,std
> kurbo v0.10.4: default,std
69d61
< memmap2 v0.2.3: 
72,76d63
< memoffset v0.7.1: default
< memoffset v0.9.0: default
< miniz_oxide v0.3.7: 
< miniz_oxide v0.4.4: no_extern_crate_alloc
< miniz_oxide v0.7.1: with-alloc
78d64
< nix v0.26.4: feature,memoffset,socket,uio,user
82,83d67
< num-rational v0.3.2: 
< num-rational v0.4.1: num-bigint,num-bigint-std,std
97,98d80
< roxmltree v0.14.1: default,std
< roxmltree v0.19.0: default,positions,std
102,105d83
< rustybuzz v0.3.0: 
< rustybuzz v0.11.0: libm,std
< siphasher v0.2.3: 
< siphasher v0.3.11: default,std
112,113d89
< toml v0.5.11: default
< toml v0.8.10: default,display,parse
121,124d96
< ttf-parser v0.9.0: default,std
< ttf-parser v0.12.3: default,std,variable-fonts
< ttf-parser v0.19.2: apple-layout,glyph-names,opentype-layout,std,variable-fonts
< ttf-parser v0.20.0: apple-layout,glyph-names,opentype-layout,variable-fonts
130,139d101
< zbus v3.15.1: async-executor,async-fs,async-io,async-lock,async-task,blocking,url
< zbus v4.0.1: async-executor,async-fs,async-io,async-lock,async-task,blocking
< zbus_macros v3.15.1 (proc-macro): 
< zbus_macros v4.0.1 (proc-macro): 
< zbus_names v2.6.1: 
< zbus_names v3.0.0: 
< zvariant v3.15.1: enumflags2,url
< zvariant v4.0.2: enumflags2,gvariant
< zvariant_derive v3.15.1 (proc-macro): 
< zvariant_derive v4.0.2 (proc-macro): 
```

</details>

Release Notes:

- N/A

niklaswimmer committed:
gpui: update ashpd and open dependency

The ashpd update removes the dependency on an older zbus version which
decreases compile times.

Signed-off-by: Niklas Wimmer <mail@nwimmer.me>

niklaswimmer committed:
gpui: make build dependencies mac only

This removes bindgen and cbindgen from the dependency graph on non-macos
systems, improving compile times on those systems.

Signed-off-by: Niklas Wimmer <mail@nwimmer.me>

niklaswimmer committed:
gpui: Update cosmic-text and resvg dependency

This unifies the rustybuzz dependency to the same version.

Signed-off-by: Niklas Wimmer <mail@nwimmer.me>

niklaswimmer committed:
gpui: Update image dependency

The latest update to resvg bumped some transitive dependencies
which lead to duplicates. The update to the image dependency
unifies most of their versions again.

Most notably, gif and kurbo are still duplicated, which is best fixed
downstream however.

Signed-off-by: Niklas Wimmer <mail@nwimmer.me>

osiewicz commented on crates/gpui/build.rs:
```diff
@@ -1,28 +1,35 @@
-    #[cfg(all(target_os = "macos", not(feature = "macos-blade")))]
-    #[cfg(not(feature = "runtime_shaders"))]
-    compile_metal_shaders(&header_path);
+    #[cfg(target = "macos")]
```
```suggestion
    #[cfg(target_os = "macos")]
```

osiewicz commented on crates/gpui/build.rs:
```diff
@@ -1,28 +1,35 @@
+
+    use cbindgen::Config;
+    
+    fn build() {
```
```suggestion
    pub(super) fn build() {
```

osiewicz commented:
This PR does not affect Mac build times significantly, so at least it's safe on that front. I'd suggest benchmarking the build timing changes with something like hyperfine, as there's a good chance that the difference you are seeing just comes down to cargo scheduler picking a more favorable crate for the build.

niklaswimmer committed:
windows: update text system to new cosmic version

The same changes have been used on linux here 500350403124a8faffc67a5162bc7745a1bac6af
and here 34832d49b09071846ff6f55f8ca1df019980a1df.

Signed-off-by: Niklas Wimmer <mail@nwimmer.me>

niklaswimmer commented:
Thanks for the feedback @osiewicz, I updated the PR description with more details

osiewicz reviewed (approved):
LGTM
